runtime/databricks/automl_runtime/forecast/utils.py

```diff
@@ -107,14 +107,18 @@ def get_validation_horizon(df: pd.DataFrame, horizon: int, unit: str) -> int:
     :param unit: frequency unit of the time series, which must be a pandas offset alias
     :return: horizon used for validation, in terms of the input `unit`
     """
-    MIN_HORIZONS = 4 # minimum number of horizons in the datafram
-    horizon_dateoffset = pd.DateOffset(**DATE_OFFSET_KEYWORD_MAP[unit])* horizon
-
-    if MIN_HORIZONS * horizon_dateoffset + df["ds"].min() <= df["ds"].max():
-        return horizon
-    else:
+    MIN_HORIZONS = 4  # minimum number of horizons in the dataframe
+    horizon_dateoffset = pd.DateOffset(**DATE_OFFSET_KEYWORD_MAP[unit]) * horizon
+
+    try:
+        validation_horizon = MIN_HORIZONS * horizon_dateoffset + df["ds"].min()
+    except OverflowError:
+        validation_horizon = None
+    finally:
```

Comment:
does this need to be in finally? I think it can be outside the try except block.

Reply:
Done